x/twap/logic.go

```diff
@@ -219,15 +227,16 @@ func (k Keeper) getMostRecentRecord(ctx sdk.Context, poolId uint64, assetA, asse
 	return record, nil
 }
 
-// computeArithmeticTwap computes and returns an arithmetic TWAP between
-// two records given the quote asset.
+// computeTwap computes and returns a TWAP of a given
+// type - arithmetic or geometric.
+// Between two records given the quote asset.
 // precondition: endRecord.Time >= startRecord.Time
 // if (endRecord.LastErrorTime >= startRecord.Time) returns an error at end + result
 // if (startRecord.LastErrorTime == startRecord.Time) returns an error at end + result
 // if (endRecord.Time == startRecord.Time) returns endRecord.LastSpotPrice
 // else returns
 // (endRecord.Accumulator - startRecord.Accumulator) / (endRecord.Time - startRecord.Time)
-func computeArithmeticTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quoteAsset string) (sdk.Dec, error) {
+func computeTwap(startRecord types.TwapRecord, endRecord types.TwapRecord, quoteAsset string, twapType twapType) (sdk.Dec, error) {
```

Comment:
In this function we short circuit if timeDelta == 0 and return `endRecord.PxLastSpotPrice`. Shouldn't that be different for Geometric twap? i.e.:  `osmomath.Pow(osmomath.Tick, endRecord.PxLastSpotPrice)`

Reply:
The idea is that if time delta from the last sample is 0, we return the sample itself.

TWAP is an estimate of the true price. If the request lands at the time of sample, the most accurate estimate is the price itself.

So I think this is what we want in both cases

Reply:
The last price is not stored in log scale. Only the accumulator is. So returning the actual price should be what we want

Reply:
ahh, of course... that makes sense.